### ISSUE TITLE
Telegram bot message stores in categorized mode

### ISSUE THREAD

**Issue by MarshalOfficial** [open-source-ideas/open-source-ideas#269]
## Project description

If possible, develop a Telegram bot, that archives any message that forward to it in categorize and get them back via search. Imagine that I start this bot and make my categories, then when I want to store a message from any-where, I use this bot to save it and easy search and get it back to use.
our problem is to make hundreds of channel for different topics or saved message is not categorized.

## Relevant Technology

Telegram API

- [*] Beginner - This project requires no or little prior knowledge of the technolog(y|ies) specified to contribute to the project
- [*] Intermediate - The user should have some prior knowledge of the technolog(y|ies) to the point where they know how to use it, but not necessarily all the nooks and crannies of the technology
- [ ] Advanced - The project requires the user to have a good understanding of all components of the project to contribute

### Required time (ETA)

- [*] Little work - A couple of days
- [*] Medium work - A week or two
- [ ] Much work - The project will take more than a couple of weeks and serious planning is required

### Categories

- [ ] Mobile app
- [ ] IoT
- [ ] Web app
- [ ] Frontend/UI
- [ ] AI/ML
- [ ] APIs/Backend
- [ ] Voice Assistant
- [ ] Developer Tooling
- [ ] Extension/Plugin/Add-On
- [ ] Design/UX
- [ ] AR/VR
- [*] Bots
- [ ] Security
- [ ] Blockchain
- [ ] Futuristic Tech/Something Unique

**Comment by Kreijstal** [comment]
Can you explain the interface of the bot, the commands?

**Comment by MarshalOfficial** [comment]
> Can you explain the interface of the bot, the commands?

Yes, sure
Scenario: If we consider that, the bot will save all messages in a single private channel:
there is a command that asks for a channel id and also client must add this bot onto that channel as an admin
there is a command that the client can define her/his categories title.
after that if possible, the client can forward the message to the bot and the bot asks for a category via button (because categories defined before and the bot knows this client's categories), and also gives a title for that message.
that title will put via hashtag, a message after original message in the store channel. why a message after? because when the bot want to search a message it will find it inside the store channel.
this is my idea, but in possibility and also for a better scenario help me.

**Comment by Kreijstal** [comment]
> > Can you explain the interface of the bot, the commands?
> 
> Yes, sure
> Scenario: If we consider that, the bot will save all messages in a single private channel:
> there is a command that asks for a channel id and also client must add this bot onto that channel as an admin
> there is a command that the client can define her/his categories title.
> after that if possible, the client can forward the message to the bot and the bot asks for a category via button (because categories defined before and the bot knows this client's categories), and also gives a title for that message.
> that title will put via hashtag, a message after original message in the store channel. why a message after? because when the bot want to search a message it will find it inside the store channel.
> this is my idea, but in possibility and also for a better scenario help me.

what would be the name of the command, like what would you type specifically?

**Comment by MarshalOfficial** [comment]
> > > Can you explain the interface of the bot, the commands?
> > 
> > 
> > Yes, sure
> > Scenario: If we consider that, the bot will save all messages in a single private channel:
> > there is a command that asks for a channel id and also client must add this bot onto that channel as an admin
> > there is a command that the client can define her/his categories title.
> > after that if possible, the client can forward the message to the bot and the bot asks for a category via button (because categories defined before and the bot knows this client's categories), and also gives a title for that message.
> > that title will put via hashtag, a message after original message in the store channel. why a message after? because when the bot want to search a message it will find it inside the store channel.
> > this is my idea, but in possibility and also for a better scenario help me.
> 
> what would be the name of the command, like what would you type specifically?

1- there will be command for defining a catergory. [Add New Category]
with this command user will create her/his categories.

2- after that scenario will be like this: me as a bot user create a channel and add this bot as admin to this channel. this channel will be my store channel that all posts will be saved there.
when i send a post or type to the bot, it ask me, what is the category of your last message, when i press for example: Learning-Docker. it then ask me what is this post title, then i type a specific titile, and finally bot send that message to channel and after that put my title and category in hashtag. to be able to retrieve message for me when search

3- the bot has 2 commands for search:
* search by category
* search by title
and will search in store channel for input value of user

**Comment by yeongjonglim** [comment]
You may want to take a look at this bot.

[SkeddyBot](https://skeddy.me/)

**Comment by MarshalOfficial** [comment]
> You may want to take a look at this bot.
> 
> [SkeddyBot](https://skeddy.me/)

It is nice bot, but it is good for saving notes and reminders, not supporting file storage solution

**Comment by name-666** [comment]
Why do you need such a bot? What business problem does it solve? What is its audience? Who will be the potential user?

**Comment by name-666** [comment]
Why do you need such a bot? What business problem does it solve? What is its audience? Who will be the potential user?

**Comment by MarshalOfficial** [comment]
> Why do you need such a bot? What business problem does it solve? What is its audience? Who will be the potential user? How do you represent the user interface?

* Users will be Telegram users who want to store their data(raw text to documents) in the Telegram secure cloud.
* Security issues will be solved here because Telegram is independent and won't share your data with NSA and other government organizations. we all know that google drive and all other social media or storage providers must and have to share users' data with the government as needed.
* Interface will be Telegram chatbot. and messenger UI

**Comment by TAKANOME-DEV** [comment]
@MarshalOfficial this is interesting, still excited about it??

**Comment by MarshalOfficial** [comment]
> @MarshalOfficial this is interesting, still excited about it??

yeah, but I don't  have enough time personally, If you are interested please start the project, I hope to join you in the near future

**Comment by PitiPipouSauvage** [comment]
> > Why do you need such a bot? What business problem does it solve? What is its audience? Who will be the potential user? How do you represent the user interface?
> 
>     * Users will be Telegram users who want to store their data(raw text to documents) in the Telegram secure cloud.
> 
>     * Security issues will be solved here because Telegram is independent and won't share your data with NSA and other government organizations. we all know that google drive and all other social media or storage providers must and have to share users' data with the government as needed.
> 
>     * Interface will be Telegram chatbot. and messenger UI

btw Telegram is basicly to the russian government

**Comment by nateweisz** [comment]
Hello there, I have decided to pick this up.

**Comment by duskwerefox** [comment]
> Hello there, I have decided to pick this up.

Greetings. Are you still working on it?

**Comment by Nopock** [comment]
> > Hello there, I have decided to pick this up.
> 
> Greetings. Are you still working on it?

I am not

**Comment by MarshalOfficial** [comment]
> > Hello there, I have decided to pick this up.
> 
> Greetings. Are you still working on it?

I am not too, but if you start may join you

**Comment by duskwerefox** [comment]
> after that if possible, the client can forward the message to the bot and the bot asks for a category via button (because categories defined before and the bot knows this client's categories), and also gives a title for that message. that title will put via hashtag, a message after original message in the store channel. why a message after? because when the bot want to search a message it will find it inside the store channel. this is my idea, but in possibility and also for a better scenario help me.

See, the thing is that API (I use python-telegram-bot) has no access to chat history, hence it cannot search for anything in a channel/group. I consider that it can be done by saving categories, titles and message IDs connected to the titles in a database.

**Comment by duskwerefox** [comment]
I'm already working on a project. Take note that message searching is not possible to implement in a direct way.
